src/main/java/com/minecolonies/coremod/client/gui/WindowBuildDecoration.java

```diff
@@ -0,0 +1,310 @@
+
+    /**
+     * List of style for the section.
+     */
+    @NotNull
+    private final List<Tuple<String, BlockPos>> builders = new ArrayList<>();
+
+    /**
+     * Drop down list for builders.
+     */
+    private DropDownList buildersDropDownList;
+
+    /**
+     * Contains all resources needed for a certain build.
+     */
+    private final Map<String, ItemStorage> resources = new HashMap<>();
+
+    /**
+     * Stores the message to be transmitted upon completion
+     */
+    private final BuildToolPlaceMessage placementMessage;
+
+    /**
+     * The name of the structure
+     */
+    private final StructureName structureName;
+
+    /**
+     * The position of the decoration anchor
+     */
+    private final BlockPos structurePos;
+
+    /**
+     * Constructs the decoration build confirmation dialog
+     */
+    public WindowBuildDecoration(BuildToolPlaceMessage msg, BlockPos pos, StructureName structure)
+    {
+        super(Constants.MOD_ID + BUILDING_NAME_RESOURCE_SUFFIX);
+        placementMessage = msg;
+        structureName = structure;
+        structurePos = pos;
+
+        registerButton(BUTTON_BUILD, this::confirmedBuild);
+        registerButton(BUTTON_CANCEL, this::close);
+
+        findPaneOfTypeByID(BUTTON_BUILD, Button.class)
+                .setLabel(LanguageHandler.format("com.minecolonies.coremod.gui.workerhuts.build"));
+        findPaneOfTypeByID(BUTTON_MOVE_BUILDING, Button.class).hide();
+        findPaneOfTypeByID(BUTTON_REPAIR, Button.class).hide();
+        findPaneOfTypeByID(BUTTON_NEXT_STYLE_ID, Button.class).hide();
+        findPaneOfTypeByID(BUTTON_PREVIOUS_STYLE_ID, Button.class).hide();
+        findPaneOfTypeByID(DROPDOWN_STYLE_ID, DropDownList.class).disable();
+    }
+
+    @Override
+    public void onOpened()
+    {
+        updateBuilders();
+        updateResources();
+    }
+
+    /**
+     * Update the builders list but try to keep the same one.
+     */
+    private void updateBuilders()
+    {
+        IColonyView colony = (IColonyView) IColonyManager.getInstance()
+                .getIColony(Minecraft.getInstance().world, structurePos);
+
+        if (colony == null)
+        {
+            LanguageHandler.sendPlayerMessage(Minecraft.getInstance().player, TranslationConstants.OUT_OF_COLONY);
+            close();
+            return;
+        }
+
+        builders.clear();
+        builders.add(new Tuple<>(LanguageHandler.format("com.minecolonies.coremod.job.Builder") + ":", BlockPos.ZERO));
+        builders.addAll(colony.getBuildings().stream()
+                .filter(build -> build instanceof AbstractBuildingBuilderView && !((AbstractBuildingBuilderView) build).getWorkerName().isEmpty()
+                        && !(build instanceof BuildingMiner.View))
+                .map(build -> new Tuple<>(((AbstractBuildingBuilderView) build).getWorkerName(), build.getPosition()))
+                .collect(Collectors.toList()));
```

Comment:
could you sort the builders by distance here? and also on the Buid Building GUI?

Reply:
I don't like doing that sort of thing simply because it is not immediately apparent when you look at the list, and the user would have to learn the order, rather than one they most likely know (the alphabet). And just IMO, the point is to allow them to select the builder, not have that choice calculated for them. That said I do understand why that would be neat and convenient, and does make sense. So I'm torn. Thoughts?